Jellyfin.Api/Controllers/ChannelsController.cs

```diff
@@ -0,0 +1,245 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChannelsController"/> class.
+        /// </summary>
+        /// <param name="channelManager">Instance of the <see cref="IChannelManager"/> interface.</param>
+        /// <param name="userManager">Instance of the <see cref="IUserManager"/> interface.</param>
+        public ChannelsController(IChannelManager channelManager, IUserManager userManager)
+        {
+            _channelManager = channelManager;
+            _userManager = userManager;
+        }
+
+        /// <summary>
+        /// Gets available channels.
+        /// </summary>
+        /// <param name="userId">User Id.</param>
+        /// <param name="startIndex">Optional. The record index to start at. All items with a lower index will be dropped from the results.</param>
+        /// <param name="limit">Optional. The maximum number of records to return.</param>
+        /// <param name="supportsLatestItems">Optional. Filter by channels that support getting latest items.</param>
+        /// <param name="supportsMediaDeletion">Optional. Filter by channels that support media deletion.</param>
+        /// <param name="isFavorite">Optional. Filter by channels that are favorite.</param>
+        /// <response code="200">Channels returned.</response>
+        /// <returns>Channels.</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult<QueryResult<BaseItemDto>> GetChannels(
+            [FromQuery] Guid userId,
+            [FromQuery] int? startIndex,
+            [FromQuery] int? limit,
+            [FromQuery] bool? supportsLatestItems,
+            [FromQuery] bool? supportsMediaDeletion,
+            [FromQuery] bool? isFavorite)
+        {
+            return _channelManager.GetChannels(new ChannelQuery
+            {
+                Limit = limit,
+                StartIndex = startIndex,
+                UserId = userId,
+                SupportsLatestItems = supportsLatestItems,
+                SupportsMediaDeletion = supportsMediaDeletion,
+                IsFavorite = isFavorite
+            });
+        }
+
+        /// <summary>
+        /// Get all channel features.
+        /// </summary>
+        /// <response code="200">All channel features returned.</response>
+        /// <returns>Channel features.</returns>
+        [HttpGet("Features")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public IEnumerable<ChannelFeatures> GetAllChannelFeatures()
+        {
+            return _channelManager.GetAllChannelFeatures();
+        }
+
+        /// <summary>
+        /// Get channel features.
+        /// </summary>
+        /// <param name="id">Channel id.</param>
+        /// <response code="200">Channel features returned.</response>
+        /// <returns>Channel features.</returns>
+        [HttpGet("{Id}/Features")]
+        public ActionResult<ChannelFeatures> GetChannelFeatures([FromRoute] string id)
```

Comment:
Looks like the swagger is case sensitive. I changed the parameter from `Id` to `channelId`. Since this is a route parameter and not a query parameter the name doesn't matter